services/auth.service.ts

```diff
@@ -56,6 +56,22 @@ async function logout(apiBase: string) {
   Router.push('/login')
 }
 
+export async function checkServerAuth(): Promise<boolean> {
+  try {
+    const res = await fetch('/api/auth/check', {
```

Comment:
Please use `await fetcher(` . This is our custom fetcher function that already handles a 401 response from the server and logs the user out

Reply:
Hey @PudgyPug, I addressed and tested changes in this commit [https://github.com/shardeum/validator-gui/pull/42/commits/23250a288ce9266dd19fe23b3b04df9265cc5bef](url)

*OpenAI is experiencing a partial outage right now. Your response may come slower, or you may have to resubmit your query. We're sorry. You can monitor the status [here](https://status.openai.com/).*

1. RouteGuard.tsx:
- Implemented `initialAuthCheck` function for one-time server auth on app load.
- Modified `authCheck` to use only client-side auth state for route changes.
- Updated state management and rendering logic for better performance.

2. auth.service.ts:
- Refactored `checkServerAuth` to use `fetcher` hook with improved error handling.